src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+
+beforeEach(() => {
+  loggerSpy = jest.spyOn(Logger, "deprecate");
+});
+
+afterEach(() => {
+  loggerSpy.mockRestore();
+});
+
+afterAll(() => {
+  loggerSpy.mockClear();
+});
+
+test("should display deprecation warning once when rendered as uncontrolled", () => {
+  render(<Textarea name="my-textarea" defaultValue="test" />);
+
+  expect(loggerSpy).toHaveBeenCalledWith(
+    "Uncontrolled behaviour in `Textarea` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+  );
+
+  expect(loggerSpy).toHaveBeenCalledTimes(1);
+});
+
+test("should render a textarea element", () => {
+  render(<Textarea />);
+  const textarea = screen.getByRole("textbox");
+  expect(textarea).toBeInTheDocument();
+  expect(textarea.tagName).toEqual("TEXTAREA");
+});
+
+test("should not render character counter if no characterLimit prop is given", () => {
+  render(<Textarea />);
+  expect(screen.queryByText(/you can enter up to/i)).not.toBeInTheDocument();
+});
+
+test("should render character counter if characterLimit prop is given", () => {
+  render(<Textarea characterLimit={5} />);
+  expect(screen.getByText(/you can enter up to/i)).toBeInTheDocument();
+});
+
+test("should not render a placeholder if disabled", () => {
+  render(<Textarea placeholder="foo" disabled />);
+  expect(screen.getByRole("textbox")).toHaveAttribute("placeholder", "");
+});
+
+testStyledSystemMargin((props) => <Textarea {...props} />);
+
+// not sure what this is really testing - I assume it's the expandTextArea stuff, but can't get it to fail!
+// would rather remove this but leave it, commented out, to see if it causes coverage problems. Would like
+// to rewrite if it does, otherwise happy to remove it!
+test("the height of the textarea should remain unchanged when the window is resized", () => {
+  render(<Textarea />);
+
+  const expectedScrollHeight = 500;
+  const originalHeight = "50px";
+  const textarea = screen.getByRole("textbox");
+
+  textarea.style.height = originalHeight;
+  jest
+    .spyOn(textarea, "scrollHeight", "get")
+    .mockImplementation(() => expectedScrollHeight);
+
+  window.dispatchEvent(new Event("resize"));
+
+  expect(textarea.style.height).toEqual(originalHeight);
+});
+
+test("should display the correct value when the value prop is set", () => {
+  render(<Textarea value="Initial content" />);
+
+  expect(screen.getByRole("textbox")).toHaveTextContent("Initial content");
+});
+
+test.each([
+  {
+    inputIcon: "search",
+  },
+  {
+    error: "error",
+  },
+  {
+    warning: "warning",
+  },
+  {
+    info: "info",
+  },
+] as const)(
+  "styles the textarea appropriately when an icon is present inside",
+  (props) => {
+    render(<Textarea value="test string" data-role="icon-test" {...props} />);
+    expect(screen.getByTestId("icon-test")).toHaveStyleRule(
+      "padding-right",
+      "var(--spacing500)",
+      { modifier: `& ${StyledInput}` }
+    );
+  }
+);
```

Comment:
Agreed. I've just been cautious throughougt and kept tests in where they were streaightforward to convert to RTL, rather than spend time on checking if style tests are already covered by Playwright/chromatic. I thought that we were going to come back and review these styling tests later - but if Chromatic definitely captures this and it's not needed for coverage (which seems unlikely in this case) then I'm happy to remove 👍

Reply:
I just checked and there's certainly no story on Textarea for the `inputIcon` prop (this prop is not even mentioned in the docs except for being included in the prop table). So to get Chromatic to capture this I'd have to add such a story - which I'm not sure is wise as it probably doesn't add very much from a documentation point of view.

Based on this I've decided to leave this test in place for now, pending a later repo-wide discussion of what to do about styling tests.